packages/core/src/shared-wallets/add-shared-wallet/generate-key-flow/EnterPassword/EnterPassword.tsx

```
@@ -44,7 +44,7 @@ export const EnterPassword: VFC<EnterPasswordProps> = ({
   const icon = mapOfWalletTypeIconProperties[kind];
 
   const next = () => {
-    onGenerateKeys(password.value);
+    onGenerateKeys(password?.value || '');
```

This may help with broken build, but are we sure that empty string is acceptable value for key generation?

This is a [place where the password is used](https://github.com/input-output-hk/lace/blob/fix/lw-11953-collateral-signing/packages/core/src/shared-wallets/add-shared-wallet/generate-key-flow/Store/machine.ts#L54) and after traversing x levels of components / abstractions etc. I discovered this [place](https://github.com/input-output-hk/lace/blob/fix/lw-11953-collateral-signing/packages/core/src/shared-wallets/add-shared-wallet/generate-key-flow/generate-shared-wallet-key.ts#L19) where password is converted to passphrase buffer and used to `getSharedWalletExtendedPublicKey`

To me it looks like empty string is not a good fit.
@szymonmaslowski would you be able to review this change, please? I see you have worked a lot on that area.

---

Maybe, but considering that it only uses an empty string when the password is empty, I don't see any issue with this. But alternatively, we could handle it like this

```
if (password.value) {
      onGenerateKeys(password.value);
      clearSecrets();
    } else {
      console.error('Password is undefined');
    }
```

---

Yes, I think we should check password has value and throw an error in case is invalid. But I'm not too familiar with this part of the codebase. Let's see what @szymonmaslowski think about this :)

Am I right that the logic should be something like this. 

1. Use password for key generation (password is required and cannot be blank)
2. If password is empty no need to generate key.

And, If password is empty, and we still try to generate key, we have some flaw in the logic? 

By passing empty string as a password do we introduce a security flaw?

---

Yes, I agree @pczeglik-iohk. We should ensure that the password is defined before generating key. 

My first approach was based on the assumption that an empty string or invalid password would throw an error during key generation.

---

Fixed on 2dce5bf

---

looks good!